### PR TITLE
Upgrade to work with unified error responses.

### PR DESCRIPTION
**Pull request by ba1ash**
Response example:
```JSON
{
  "response" :
    {
      "message":"........",
      "errors":
        {
          "one":"something go wrong with one",
          "two":"something go wrong with two"
        }
    }
}
```

### CODE REVIEW

**Comment by dm1try** [comment]
:-1: any error response will be empty if these changes will be applied

**Comment by dm1try** [comment]
the failing spec
```ruby
  context 'Faraday client raises an error' do
    before do
      allow_any_instance_of(Faraday::Connection).to receive(:public_send).and_raise(Faraday::Error::TimeoutError)
    end

    it 'returns an error response' do
      response = checkout.get_token({})

      expect(response.invalid?).to eq(true)
      expect(response.message).to eq('Gateway is temporarily unavailable')
    end
  end
```

**Comment by dm1try** [comment]
`#send_request` smells bad, it returns objects with different types for positive/negative(rescued) branches:
```ruby
      begin
        connection.public_send(method, path, params) # here we return some Faraday response returned
      rescue Faraday::Error::ClientError => e
         ErrorResponse.new(...) # here we return some BeGateway::ErrorResponse 
      end
    end
```
The client code does not catch this, so if you write some client code(was popped from `Checkout`):
```ruby
      response = post "/ctp/api/checkouts", { checkout: params }
      # here the response can be FaradayResponse or BegatewayErrorResponse
      # but they have different interfaces
      make_response(response)
      # make_response works only with Faraday response(it needs `#status` and `#body`)
```
it will fail silently (am empty response because of using OpenStruct.. btw)

Solution:
  in any case return from `send_request` the instance which implements `SomethingWithBodyAndStatus` interface.

**Comment by ba1ash** [comment]
:up: 
Thanks, @dm1try !

**Comment by dm1try** [comment]
looks legit for me :sunglasses:

**Comment by dm1try** [comment]
Anyway the general problem (it's not related to this issue) that `OpenStruct` is overused in places where the explicit type should be defined. 
OpenStruct breaks the "fail-fast" principle. It responds to any "undefined" method with `nil` (in ruby world, it "quacks" for anything). ex:
```ruby
(200..299).include?(response.status) # works fine for any openstruct and does not fail fast
```
This extends the scope of a problem. 
Good luck!

File: lib/be_gateway/response/error_response.rb [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
 
     def invalid?
       true
     end
-    
+
     def transaction_type
-      transaction["type"]
+      transaction['type']
     end
```
Comment (seongreen) [inline lib/be_gateway/response/error_response.rb]: Went we can met a "transaction" in error response ?

Comment (ba1ash) [inline lib/be_gateway/response/error_response.rb]: I think we cannot meet it. Therefore, we can remove this method with no problem. Do you agree?

Comment (seongreen) [inline lib/be_gateway/response/error_response.rb]: :point_right:

**Comment by shiroginne** [comment]
I'll merge this PR and add 'feature' issue: change OpenStruct -> proper Class, thank you @dm1try for pointing.

**Pull request merged**